capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -179,7 +173,7 @@ public boolean putPageView(String restaurantKey) {
                 // Query for the name
                 Query nameQuery = new Query("RestaurantInfo")
                         .setFilter(new FilterPredicate(
-                                "restaurantKey", Query.FilterOperator.EQUAL, restaurantKey));
+                                Entity.KEY_RESERVED_PROPERTY, Query.FilterOperator.EQUAL, KeyFactory.createKey("RestaurantInfo", Long.parseLong(restaurantKey))));
```

Comment:
Are we certain that the only data encoded within a entity's key is it's `kind` and it's ID? From my understanding, IDs are only unique to an entity `kind` when Datastore creates it. This fact, in conjunction with the fact that one could theoretically create their own not-necessarily-unique ID system leads me to believe that an entity's key is not uniquely defined by its `kind` and its ID.

Reply:
What do you mean exactly? I tried testing everything all together and it seemed to be correctly populating metrics for the restaurants. Was your worry that this query would not be able to find the correct RestaurantInfo entity or that it might find duplicates because the Key it creates won't be unique? 
For reference, this is where I found the createKey method currently being used: https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/KeyFactory.html#createKey-java.lang.String-long-

Reply:
My concern had less to do with the uniqueness of the entity keys and more to do with the uniqueness of the IDs of the keys. If IDs are not required to be unique within an entity `kind`, then a relation from `(kind, ID)` to entity key would not necessarily be functional in nature, if that makes sense. Essentially, you wouldn't be guaranteed that a specific key is returned given a `kind` and an ID. So if there were two entities of the same `kind`, with the same ID, they would have two different keys, which is problematic because your method of creating a key wouldn't guarantee the correct key is made. It turns out that none of that matters because the IDs are required to be unique for all root entities of a specific kind, and we only use root entities. Your method should be fine, then.

Reply:
Ah I see, thanks so much for looking into that to make sure it'll work, I appreciate it!